helix-term/src/application.rs

```diff
@@ -280,19 +280,13 @@ impl Application {
                     ..
                 }) => {
                     debugger.is_running = false;
-                    let main = debugger.threads().await.ok().and_then(|threads| {
-                        // Workaround for debugging Go tests. Main thread has * in beginning of its name
-                        let mut main = threads.iter().find(|t| t.name.starts_with('*')).cloned();
-                        if main.is_none() {
-                            main = threads.get(0).cloned();
-                        }
-                        main
-                    });
 
-                    if let Some(main) = main {
-                        let (bt, _) = debugger.stack_trace(main.id).await.unwrap();
+                    // whichever thread stops is made "current".
+                    debugger.thread_id = thread_id;
```

Comment:
Remember about `allThreadsStopped`

Reply:
Both nvim and emacs seem to do it this way:

https://github.com/mfussenegger/nvim-dap/blob/42e1eec29d14a2ed159cffcf1b3c0d1b6018b855/lua/dap/session.lua#L275

https://github.com/emacs-lsp/dap-mode/blob/8ec7e98986ea46e3c36b7ecbdf9a6562c90c7632/dap-mode.el#L942

Though emacs will only change the currently active thread_id if one wasn't already set before.